heron/schedulers/src/java/org/apache/heron/scheduler/aurora/AuroraScheduler.java

```diff
@@ -279,6 +279,19 @@ public void removeContainers(Set<PackingPlan.ContainerPlan> containersToRemove)
     auroraProperties.put(AuroraField.CORE_PACKAGE_URI, heronCoreReleasePkgURI);
     auroraProperties.put(AuroraField.TOPOLOGY_PACKAGE_URI, topologyPkgURI);
 
+    if (config.containsKey(Key.SCHEDULER_PROPERTIES)) {
```

Comment:
Still not idea to have AURORA_METADATA_1, AURORA_METADATA_2, and AURORA_METADATA_3.

I am thinking maybe in addition to auroraProperties, which is for standard properties with type Map[AuroraField, String], we can add another Map[String, String] extraProperties for these dynamic properties, so that we can avoid the hard coded names.

Reply:
if i understand correctly, the new 'Map[AuroraField, String]' will be added in the method param list https://github.com/apache/incubator-heron/blob/0.17.8/heron/schedulers/src/java/com/twitter/heron/scheduler/aurora/AuroraController.java#L26. Thoughts to update the interface? @nlu90 @maosongfu

Reply:
The new map would be the new 'Map[String, String]'. No need to add to param list I think since the values are generated from existing arguments, but I might be wrong.

Reply:
nvm. i see it now.

Reply:
after discussion, i am going to update the method to `boolean createJob(Map<AuroraField, String> auroraProperties, Map<String, String> extra);`